packages/react-router/lib/router/define-route.ts

```diff
@@ -75,13 +73,13 @@ type _LoaderData<
   ClientLoaderHydrate extends boolean,
   HydrateFallback,
 > = Awaited<
-  [IsDefined<HydrateFallback>, ClientLoaderHydrate]  extends [true, true] ?
+  [undefined extends HydrateFallback ? false : true, ClientLoaderHydrate]  extends [true, true] ?
```

Comment:
Unlike for `Data` types (e.g. `ServerLoaderData`), its correct to check for `undefined extends` for `HydrateFallback` since `HydrateFallback` is a top-level value of `defineRoute` whereas the things like `ServerLoaderData` are inferred from the return type of a top-level value like `serverLoader`.